src/network/networkd-ndisc.c

```diff
@@ -149,6 +150,290 @@ static int ndisc_check_ready(Link *link) {
+                        return true;
+
+        ORDERED_SET_FOREACH(req, link->manager->request_queue) {
+                if (req->type != REQUEST_TYPE_NEXTHOP)
+                        continue;
+
+                if (ndisc_take_nexthop_id(nexthop, req->userdata, link->manager))
+                        return true;
+        }
+
+        return false;
+}
+
+static int ndisc_nexthop_new(Route *route, Link *link, NextHop **ret) {
+        _cleanup_(nexthop_unrefp) NextHop *nexthop = NULL;
+        int r;
+
+        assert(route);
+        assert(link);
+        assert(ret);
+
+        r = nexthop_new(&nexthop);
+        if (r < 0)
+                return r;
+
+        nexthop->source = NETWORK_CONFIG_SOURCE_NDISC;
+        nexthop->provider = route->provider;
+        nexthop->protocol = route->protocol == RTPROT_REDIRECT ? RTPROT_REDIRECT : RTPROT_RA;
+        nexthop->family = AF_INET6;
+        nexthop->gw.address = route->nexthop.gw;
+        nexthop->ifindex = link->ifindex;
+
+        r = ndisc_nexthop_find_id(nexthop, link);
+        if (r < 0)
+                return r;
+
+        *ret = TAKE_PTR(nexthop);
+        return 0;
+}
+
+static int ndisc_nexthop_acquire_id(NextHop *nexthop, Link *link) {
+        int r;
+
+        assert(nexthop);
+        assert(nexthop->id == 0);
+        assert(link);
+        assert(link->manager);
+
+        sd_id128_t app_id;
+        r = sd_id128_get_machine_app_specific(NDISC_NEXTHOP_APP_ID, &app_id);
+        if (r < 0)
+                return r;
+
+        for (uint64_t trial = 0; trial < 100; trial++) {
+                uint32_t id = ndisc_generate_nexthop_id(nexthop, link, app_id, trial);
+                if (id == 0)
+                        continue;
+
+                if (set_contains(link->manager->nexthop_ids, UINT32_TO_PTR(id)))
+                        continue; /* The ID is already used in a .network file. */
+
+                if (nexthop_get_by_id(link->manager, id, NULL) >= 0)
+                        continue; /* The ID is already used by an existing nexthop. */
+
+                if (nexthop_get_request_by_id(link->manager, id, NULL) >= 0)
+                        continue; /* The ID is already used by a nexthop being requested. */
+
+                log_link_debug(link, "Generated new ndisc nexthop ID for %s with trial %"PRIu64": %"PRIu32,
+                               IN6_ADDR_TO_STRING(&nexthop->gw.address.in6), trial, id);
+                nexthop->id = id;
+                return 0;
+        }
+
+        return log_link_debug_errno(link, SYNTHETIC_ERRNO(EBUSY), "Cannot find free nexthop ID for %s.",
+                                    IN6_ADDR_TO_STRING(&nexthop->gw.address.in6));
+}
+
+static int ndisc_nexthop_handler(sd_netlink *rtnl, sd_netlink_message *m, Request *req, Link *link, NextHop *nexthop) {
+        int r;
+
+        assert(link);
+
+        r = nexthop_configure_handler_internal(m, link, "Could not set NDisc route");
```

Comment:
Looking at
```sh
host0: Trying to reconfigure the interface.
host0: Could not set NDisc route: Connection timed out
host0: Failed
host0: The interface entered the failed state frequently, refusing to reconfigure it automatically.
```
it seems network interfaces go down here somehow.

(I don't think I have seen it before)

There is also a memory leak: https://github.com/systemd/systemd/issues/35164

Reply:
The memory leak is getting bigger:
```sh
==559==ERROR: LeakSanitizer: detected memory leaks
Direct leak of 5248 byte(s) in 41 object(s) allocated from:
    #0 0x7f936e4d5aa3 in malloc (/usr/lib/clang/19/lib/x86_64-redhat-linux-gnu/libclang_rt.asan.so+0xd5aa3) (BuildId: 98d6f0ad935c4ea5ed3cdc909871a71680370a1>
    #1 0x55871f331d8d in malloc_multiply /work/build/../src/src/basic/alloc-util.h:119:16
    #2 0x55871f33190b in nexthop_new /work/build/../src/src/network/networkd-nexthop.c:128:19
    #3 0x55871f345902 in nexthop_add_new /work/build/../src/src/network/networkd-nexthop.c:412:13
    #4 0x55871f342c3f in manager_rtnl_process_nexthop /work/build/../src/src/network/networkd-nexthop.c:1140:21
    #5 0x7f936dc4f4f1 in process_match /work/build/../src/src/libsystemd/sd-netlink/sd-netlink.c:314:21
    #6 0x7f936dc453ce in process_running /work/build/../src/src/libsystemd/sd-netlink/sd-netlink.c:344:21
    #7 0x7f936dc44f71 in sd_netlink_process /work/build/../src/src/libsystemd/sd-netlink/sd-netlink.c:374:13
    #8 0x7f936dc4a078 in io_callback /work/build/../src/src/libsystemd/sd-netlink/sd-netlink.c:655:13
    #9 0x7f936d8a2b95 in source_dispatch /work/build/../src/src/libsystemd/sd-event/sd-event.c:4228:21
    #10 0x7f936d8a0c18 in sd_event_dispatch /work/build/../src/src/libsystemd/sd-event/sd-event.c:4849:21
    #11 0x7f936d8a6566 in sd_event_run /work/build/../src/src/libsystemd/sd-event/sd-event.c:4910:21
    #12 0x7f936d8a74ff in sd_event_loop /work/build/../src/src/libsystemd/sd-event/sd-event.c:4931:21
    #13 0x55871efc6c8d in run /work/build/../src/src/network/networkd.c:114:13
    #14 0x55871efc3d97 in main /work/build/../src/src/network/networkd.c:121:1
    #15 0x7f936c611247 in __libc_start_call_main (/lib64/libc.so.6+0x3247) (BuildId: 948d229348b56bdcf4e0a96d5ccd8fb175f617f0)
    #16 0x7f936c61130a in __libc_start_main@GLIBC_2.2.5 (/lib64/libc.so.6+0x330a) (BuildId: 948d229348b56bdcf4e0a96d5ccd8fb175f617f0)
    #17 0x55871efc3b44 in _start (/usr/lib/systemd/systemd-networkd+0x15fb44) (BuildId: ab2c9531981cbfe2507953d7e4a50ec991e70568)
SUMMARY: AddressSanitizer: 5248 byte(s) leaked in 41 allocation(s).
systemd-networkd.service: Main process exited, code=exited, status=1/FAILURE
```
If all goes well I'll recover RAs triggering all that.